src/AutoUI/utils.ts

```diff
@@ -72,22 +77,27 @@ export const ObjectFromEntries = (entries: any[]) => {
 	return obj;
 };
 
-export const getTagsDisabledReason = <T extends AutoUIBaseResource<T>>(
+export const getTagsDisabledReason = async <T extends AutoUIBaseResource<T>>(
 	selected: T[] | undefined,
 	tagField: keyof T,
+	checkedState: CheckedState | undefined,
+	tagsSdk: AutoUITagsSdk<T>,
 	t: TFunction,
 ) => {
-	if (!selected || selected.length === 0) {
+	if (checkedState !== 'all' && (!selected || selected.length === 0)) {
 		return t('info.no_selected');
 	}
 
-	const lacksPermissionsOnSelected = selected.some((entry) => {
-		return (
-			!entry.__permissions.delete &&
-			!entry.__permissions.create.includes(tagField) &&
-			!entry.__permissions.update.includes(tagField as keyof T)
-		);
-	});
+	const lacksPermissionsOnSelected =
+		checkedState === 'all' && tagsSdk && 'canAccess' in tagsSdk
+			? await tagsSdk.canAccess()
```

Comment:
Should we also have a case when `chekedState !== 'all'` but we are still in server side pagination mode and in that case do `tagsSdk.canAccess(pineFilter)`?

Reply:
We could, I was thinking to keep it simpler and use the same rules for all other selections as in the manual selection we still have entities with _permissions on it. Also, this would allow us to save a couple of unnecessary API requests  . WDYT ?

Reply:
I'm fine with it, but in that case how about changing the signature to:
```
canAccess: (param: 'all' | T[]) => Promise<boolean>;
```
This would allow us to use `canAccess` even on client side more and then this code becomes:
```
const lacksPermissionsOnSelected =
		tagsSdk != null && 'canAccess' in tagsSdk
			? await tagsSdk.canAccess(checkedState === 'all' ? 'all' : selected)
			: selected?.some((entry) => ...)
```

Reply:
To be clear @JSReds I mean that it feel weird to have a `canAccess()` parameter to control permissions, but then it is only called when ALL items are selected and it isn't when I just select 1-2. As a consumer I would be puzzled of it not being called. It might even bite us later, since the authority of "who checks the permissions & how" will now be split in two between the UI & autoUI.

Semantically I would prefer if it was always called when provided, and leave optimizations such as using  `__permissions` to the UI/consumer to implement on their side.

Reply:
```
canAccess: (param: { selectionState?: 'all'; selected?: T[] }) => Promise<boolean>;
```